### ISSUE TITLE
Transfer My Schedule between devices

### ISSUE THREAD

**Issue by lostcarpark** [lostcarpark/conclar#64]
Allow for people to transfer their personal schedule between devices. This becomes especially useful if custom notes or personal appointments are supported. Many people would find it convenient to set up their schedule on a Desktop at home or laptop in their hotel room, then transfer to their phone or tablet to refer to at the convention.

Several possible approaches:

1. Add "download/upload" buttons, that allow a file to be saved (I'd vote for JSON format), and loaded on another device. Personally, I think this would be useful for external editing, or for people who want a copy of the data that can be referred after the app is no longer online. But transferring files to mobile devices is a pain, so I don't think this would be the main method.
2. Share by URL. Encode the "My Schedule" data in the URL, and following the link opens ConClár and loads the schedule. One problem with this is that while FireFox allows almost unlimited URLs, Chrome has a maximum URL length of 2048 bytes. That would be plenty for storing selections, but would be quite limiting for personal notes. This could possibly be got around by providing several URLs that people could click in turn, each adding part of their schedule to localStorage.
3. Share by QR code. It's easy to bake a QR code from a React app. It would be fairly simple to take the URL from 2 and encode as a QR code that people could scan on their phone, and it would take them into ConClár with their schedule. QR codes have a maximum storage of about 4K, which is more than Chrome's longest URL, so wouldn't be a concern.
4. Share by API. The idea would be that members could log into a back end and their selections and notes would automatically transfer between the devices they log in from. This could potentially be extended to allow people to give "friends" access to their schedule or invite people to personal events. It could also be used for sign-up for limited attendance events like Café Klatches.

I think there's potentially scope for including all of these methods, and they would all have their uses.

I think the API option could get very interesting. However, I think any back end should be treated as a separate project from ConClár. I'd prefer to see a well defined API specification that back end application could implement. This could allow it to be tied into existing registration or programming planning systems (there are arguments for either or both being used for back ends).

**Comment by aJanuary** [comment]
@lostcarpark I think 3 is fully implemented and 4 is mostly implemented. I think this issue can be closed?

**Comment by lostcarpark** [comment]
Yeah. I think 2 is also covered, as the link the QR code uses is also shared on the "My Profile" page.
